domain/utxoindex/utxoindex.go

```diff
@@ -93,6 +93,36 @@ func (ui *UTXOIndex) isSynced() (bool, error) {
 	return externalapi.HashesEqual(virtualInfo.ParentHashes, utxoIndexVirtualParents), nil
 }
 
+func (ui *UTXOIndex) UpdateUtxoDiff(utxoDiff externalapi.UTXODiff) (*UTXOChanges, error) {
+	ui.mutex.Lock()
+	defer ui.mutex.Unlock()
+
+	log.Tracef("Updating UTXO index with VirtualUTXODiff: %+v", utxoDiff)
```

Comment:
You don't update the virtual parents in this method. IMHO you should make `ResolveVirtual` return the virtual parents and then just call `Update`